core/src/services/webhdfs/writer.rs

```diff
@@ -41,14 +43,22 @@ impl WebhdfsWriter {
 }
 
 #[async_trait]
-impl oio::OneShotWrite for WebhdfsWriter {
+impl oio::BlockWrite for WebhdfsWriter {
     /// Using `bytes` instead of `vectored_bytes` to allow request to be redirected.
-    async fn write_once(&self, bs: &dyn WriteBuf) -> Result<()> {
-        let bs = bs.bytes(bs.remaining());
+    async fn write_once(&self, size: u64, body: AsyncBody) -> Result<()> {
+        let bs = match body {
```

Comment:
:laughing:, why not passing `body` directly. They are exactly the same type.

Reply:
Using `bytes` instead of `vectored_bytes` to allow request to be redirected.

Reply:
Ah, got it. We can introduce `noredirect=true` for all API in the future.